impl/src/main/java/org/jboss/weld/annotated/enhanced/jlr/EnhancedAnnotatedTypeImpl.java

```diff
@@ -345,10 +342,8 @@ protected EnhancedAnnotatedTypeImpl(SlimAnnotatedType<T> annotatedType,
             Set<EnhancedAnnotatedMethod<?, ? super T>> effectiveMethods) {
         Multimap<Class<? extends Annotation>, EnhancedAnnotatedMethod<?, ? super T>> result = SetMultimap.newSetMultimap();
         for (EnhancedAnnotatedMethod<?, ? super T> method : effectiveMethods) {
-            for (Class<? extends Annotation> annotation : MAPPED_METHOD_ANNOTATIONS) {
-                if (method.isAnnotationPresent(annotation)) {
-                    result.put(annotation, method);
-                }
+            for (Annotation ann : method.getAnnotations()) {
+                result.put(ann.annotationType(), method);
```

Comment:
This change is not needed for the PR but fixes the inconsistency in the code - previously the collection of annotated methods only contained methods with certain annotations. This changes it to instead know about all annotated methods in given AT - which is what the API javadoc for `EnhancedAnnotatedType#getEnhancedMethods(Class)` states to begin with